### PR TITLE
Hide the core backup integration if hassio is loaded

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Allow defining components that if present, should not show the entries.

Needed for <https://github.com/home-assistant/core/pull/110651>

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/layouts/hass-tabs-subpage.ts [hunk 1]
```diff
@@ -26,6 +26,8 @@ export interface PageNavigation {
   component?: string;
   components?: string[];
   name?: string;
+  not_component?: string;
+  not_components?: string[];
```
Comment (ludeeus) [inline src/layouts/hass-tabs-subpage.ts:30]: I'm open to name suggestions here, if there are any other alternatives that make more sense.

**Comment by bramkragten** [comment]
Would this be temporary? Until the supervisor integration implements the backup platform?

File: src/layouts/hass-tabs-subpage.ts (hunk 1 above)
Comment (bramkragten) [reply src/layouts/hass-tabs-subpage.ts:30]: we should just combine them 🙈 
`not_component?: string | string[];`

```js
Array.isArray(not_component) ? page.not_component.some((integration) => isComponentLoaded(hass, integration)) : isComponentLoaded(hass, page.not_component)
```
OR
```js
ensureArray(not_component).some((integration) => isComponentLoaded(hass, integration));
```

Comment (ludeeus) [reply src/layouts/hass-tabs-subpage.ts:30]: Was just copying the current `component(s)` entries

Comment (bramkragten) [reply src/layouts/hass-tabs-subpage.ts:30]: Yeah I know, they suck too 😄

**Review by bramkragten — DISMISSED** [review]

**Review by bramkragten — APPROVED** [review]

File: src/layouts/hass-tabs-subpage.ts (hunk 1 above)
Comment (ludeeus) [reply src/layouts/hass-tabs-subpage.ts]: https://github.com/home-assistant/frontend/pull/19890

**Pull request merged**